### DISCUSSION TITLE
Lesspipe 2.16 Error On Usage

### DISCUSSION THREAD

**Discussion by bimmerite** [Homebrew/discussions#5736] [Tap maintenance and Homebrew development]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 254bf3fe9d8fa2e1b2fb55dbcf535b2d870180c4
Last commit: 5 days ago
Core tap JSON: 15 Nov 21:24 UTC
Core cask tap JSON: 15 Nov 21:24 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 3.3.6 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 16.0.0 build 1600
Git: 2.39.5 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.7.1-x86_64
CLT: 16.1.0.0.1.1729049160
Xcode: 16.1
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  hping

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/Spyder.dylib
  /usr/local/lib/Spyder3.dylib
  /usr/local/lib/Spyder5.dylib
  /usr/local/lib/dccmtr.dylib
  /usr/local/lib/dcxmtr.dylib
  /usr/local/lib/libCIEHandler.dylib
  /usr/local/lib/libCalibrationHandler.dylib
  /usr/local/lib/libDeviceManager.dylib
  /usr/local/lib/libErrorHandler.dylib
  /usr/local/lib/libICMManager.dylib
  /usr/local/lib/libMCCAppManager.dylib
  /usr/local/lib/libMCCEngineWrapper.dylib
  /usr/local/lib/libMonitorDDCCISDK.dylib
  /usr/local/lib/libProtocolEngine.dylib
```

### Description of issue

MacOS 14.7.1.  Below is the 'Brew info lesspipe' output:
==> lesspipe: stable 2.16 (bottled)
Input filter for the pager less
https://www-zeuthen.desy.de/~friebel/unix/lesspipe.html
Installed
/usr/local/Cellar/lesspipe/2.16 (15 files, 215KB) *
  Poured from bottle using the formulae.brew.sh API on 2024-11-14 at 13:13:56
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/l/lesspipe.rb
License: GPL-2.0-only AND GPL-2.0-or-later AND MIT AND (Artistic-1.0-Perl OR GPL-1.0-or-later)

When running less, I received the following error:
 $ less .zshenv
/usr/local/bin/lesspipe.sh: line 844: declare: -l: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]

I found the bug that was reported back in 2023 in the lesspipe git with the solution being to run the lesspipe.sh file using zsh instead of bash due to limitations in MacOS BASH implementation.

When I checked the installed copy of lesspipe.sh I found that it was still calling bash instead of zsh:  

$ vim /usr/local/bin/lesspipe.sh
From the lesspipe.sh file:
#!/usr/bin/env bash

I changed this to:  #!/usr/bin/env zsh
And the error went away.

Was the brew install of lesspipe supposed to be updated to call the zsh shell instead of the bash shell?

**Comment by gromgit** [comment]
I think a better solution is to have `lesspipe` depend on Homebrew `bash` instead. Since the developers are clearly bash-centric, further development may introduce bashisms that don't work in Zsh.

**Reply by gromgit** [reply]
@bimmerite [PR](https://github.com/Homebrew/homebrew-core/pull/197880) has been merged. `brew update` and `brew reinstall lesspipe`, then try again.

**Comment by bimmerite** [comment]
This worked like a charm.  Saw that reinstalling lesspipe installed bash dependencies.  Once done I re-ran less and no more error.
Thank you for such a fast response and fix!
Homebrew is the best!!
